libflux/flux-core/src/semantic/types.rs

```diff
@@ -2189,8 +2189,10 @@ impl Function {
 pub(crate) trait TypeLike {
     type Error;
     fn typ(&self) -> &MonoType;
+    #[allow(dead_code)]
     fn into_type(self) -> MonoType;
     fn error(&self, error: Error) -> Self::Error;
+    #[allow(dead_code)]
     fn location(&self) -> crate::ast::SourceLocation;
```

Comment:
Is there a reason to keep these methods around in the trait definition if they're not used anywhere? I have zero context for this trait, so feel free to ignore me.

Reply:
Mainly because I didn't want to mess with the code any more than I have to. I don't really know much about how this library works, other than it having a scarily brittle build process for interfacing with the go part of the system.